source/server/listener_impl.cc

```diff
@@ -679,9 +679,21 @@ void ListenerImpl::buildSocketOptions() {
 #else
     // Not in place listener update.
     if (config_.has_connection_balance_config()) {
-      // Currently exact balance is the only supported type and there are no options.
-      ASSERT(config_.connection_balance_config().has_exact_balance());
-      connection_balancer_ = std::make_shared<Network::ExactConnectionBalancerImpl>();
+      if (config_.connection_balance_config().has_exact_balance()) {
```

Comment:
This should still be a switch so we know if something new is added.

Reply:
I am not sure, if something new has been added, there are still 3 cases: exact, extend, and nop. All new things are covered in extend by type_url.

Reply:
I changed it to a switch.